shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -1221,6 +1223,19 @@ public void setViewportMetrics(@NonNull ViewportMetrics viewportMetrics) {
       displayFeaturesType[i] = displayFeature.type.encodedValue;
       displayFeaturesState[i] = displayFeature.state.encodedValue;
     }
+    int cutoutOffset = viewportMetrics.displayFeatures.size() * 4;
+    for (int i = 0; i < viewportMetrics.displayCutouts.size(); i++) {
+      DisplayCutout displayCutout = viewportMetrics.displayCutouts.get(i);
+      displayFeaturesBounds[cutoutOffset + 4 * i] = displayCutout.bounds.left;
+      displayFeaturesBounds[cutoutOffset + 4 * i + 1] = displayCutout.bounds.top;
+      displayFeaturesBounds[cutoutOffset + 4 * i + 2] = displayCutout.bounds.right;
+      displayFeaturesBounds[cutoutOffset + 4 * i + 3] = displayCutout.bounds.bottom;
```

Comment:
Consider authoring a helper here since this code is repeated and any bugs in one would likely be a bug in the other.